pkg/sdk/testint/external_volumes_gen_integration_test.go

```diff
@@ -0,0 +1,605 @@
+					panic("Unrecognised storage provider in storage location property")
+				}
+			} else {
+				externalVolumePropNameValue = append(externalVolumePropNameValue, ExternalVolumePropNameValue{Name: p.Name, Value: p.Value})
+			}
+		}
+
+		return externalVolumePropNameValue
+	}
+
+	// Storage location structs for testing
+	// Note cannot test awsgov on non-gov Snowflake deployments
+
+	s3StorageLocations := []sdk.ExternalVolumeStorageLocation{
+		{
+			S3StorageLocationParams: &sdk.S3StorageLocationParams{
+				Name:                 "s3_testing_storage_location",
+				StorageProvider:      sdk.S3StorageProviderS3,
+				StorageAwsRoleArn:    awsRoleARN,
+				StorageBaseUrl:       awsBaseUrl,
+				StorageAwsExternalId: sdk.String(awsExternalId),
+				Encryption: sdk.ExternalVolumeS3Encryption{
+					Type:     sdk.S3EncryptionTypeSseKms,
+					KmsKeyId: &awsKmsKeyId,
+				},
+			},
+		},
+	}
+
+	s3StorageLocationsNoEncryption := []sdk.ExternalVolumeStorageLocation{
+		{
+			S3StorageLocationParams: &sdk.S3StorageLocationParams{
+				Name:                 "s3_testing_storage_location_no_encryption",
+				StorageProvider:      sdk.S3StorageProviderS3,
+				StorageAwsRoleArn:    awsRoleARN,
+				StorageBaseUrl:       awsBaseUrl,
+				StorageAwsExternalId: sdk.String(awsExternalId),
+				Encryption: sdk.ExternalVolumeS3Encryption{
+					Type: sdk.S3EncryptionNone,
+				},
+			},
+		},
+	}
+
+	gcsStorageLocationsNoEncryption := []sdk.ExternalVolumeStorageLocation{
+		{
+			GCSStorageLocationParams: &sdk.GCSStorageLocationParams{
+				Name:           "gcs_testing_storage_location_no_encryption",
+				StorageBaseUrl: gcsBaseUrl,
+				Encryption: sdk.ExternalVolumeGCSEncryption{
+					Type: sdk.GCSEncryptionTypeNone,
+				},
+			},
+		},
+	}
+
+	gcsStorageLocations := []sdk.ExternalVolumeStorageLocation{
+		{
+			GCSStorageLocationParams: &sdk.GCSStorageLocationParams{
+				Name:           "gcs_testing_storage_location",
+				StorageBaseUrl: gcsBaseUrl,
+				Encryption: sdk.ExternalVolumeGCSEncryption{
+					Type:     sdk.GCSEncryptionTypeSseKms,
+					KmsKeyId: &gcsKmsKeyId,
+				},
+			},
+		},
+	}
+
+	azureStorageLocations := []sdk.ExternalVolumeStorageLocation{
+		{
+			AzureStorageLocationParams: &sdk.AzureStorageLocationParams{
+				Name:           "azure_testing_storage_location",
+				AzureTenantId:  azureTenantId,
+				StorageBaseUrl: azureBaseUrl,
+			},
+		},
+	}
+
+	createExternalVolume := func(t *testing.T, storageLocations []sdk.ExternalVolumeStorageLocation, allowWrites bool, comment string) sdk.AccountObjectIdentifier {
+		t.Helper()
+
+		id := testClientHelper().Ids.RandomAccountObjectIdentifier()
+		req := sdk.NewCreateExternalVolumeRequest(id, storageLocations).
+			WithIfNotExists(true).
+			WithAllowWrites(allowWrites).
+			WithComment(comment)
+
+		err := client.ExternalVolumes.Create(ctx, req)
+		require.NoError(t, err)
+
+		t.Cleanup(func() {
+			err := client.ExternalVolumes.Drop(ctx, sdk.NewDropExternalVolumeRequest(id))
```

Comment:
Please add `if exists` here - usually it does not matter but in a few setups it helps (e.g. test where you want to prematurely remove the object during the test run - with if exists the initially initiated cleanup won't result in an error; e.g. we have rename tests that benefit from it)